src/snowflake/snowpark/_internal/udf_utils.py

```diff
@@ -806,8 +869,66 @@ def {_DEFAULT_HANDLER_NAME}({wrapper_params}):
 """.strip()
 
 
+def add_snowpark_package_to_sproc_packages(
+    session: Optional["snowflake.snowpark.Session"],
+    packages: Optional[List[Union[str, ModuleType]]],
+) -> List[Union[str, ModuleType]]:
+    major, minor, patch = VERSION
+    package_name = "snowflake-snowpark-python"
+    # Use == to ensure that the remote version matches the local version
+    this_package = f"{package_name}=={major}.{minor}.{patch}"
+
+    # When resolve_imports_and_packages is called below it will use the provided packages or
```

Comment:
Let's put these comments in a docstring.

Reply:
Discussed on zoom, these comments help understand flow of the code, but docstrings help with overall description and behavior of the function. Will leave as inline comments.